src/auth/auth-store.tsx

```diff
@@ -14,11 +14,14 @@ interface RegistrationCredentials extends Credentials {
 interface AuthState {
   session: Session | null | undefined;
   isLoggedIn: () => boolean | undefined;
+  getUserData: () => User | undefined;
   login: ({ email, password }: Credentials) => void;
```

Comment:
login, register, updatePassword and some more are async and should return `Promise<void>`